File: liqp-core/src/main/java/liqp/filters/Uniq.java
package liqp.filters;

import java.util.Arrays;
import java.util.LinkedHashSet;
import java.util.Set;
import liqp.nodes.RenderContext;

public class Uniq extends Filter {

    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        if (!super.isArray(value)) {
            return value;
        }

        Set<Object> set = new LinkedHashSet<Object>(Arrays.asList(super.asArray(value)));

        return set.toArray();
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Uniq.kt
package liqp.filter

import java.util.LinkedHashSet
import liqp.context.LContext
import liqp.params.FilterParams

class Uniq : LFilter() {

  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return LinkedHashSet(context.asIterable(value).toList()).toList()
  }
}
